source/NVDAObjects/IAccessible/__init__.py

```diff
@@ -1663,9 +1666,14 @@ def _get_role(self):
 	def _get_name(self):
 		# #4364 On some versions of Windows name contains redundant information that is available either in the role or the value, however on Windows 10 Anniversary Update and later the value is empty, so we cannot simply dismiss the name.
 		if super(TrayClockWClass, self).value is None:
-			return super(TrayClockWClass, self).name
+			return super(TrayClockWClass, self).name.replace(CHAR_LTR_MARK,'').replace(CHAR_RTL_MARK,'')
 		return None
 
+	def _get_value(self):
+		if super(TrayClockWClass, self).value is None:

```

Comment:
Again, I would collect the value in its own local variable, perform the replacement only if it is not NOne and finally return the variable.